rosetta-ide/src/main/java/com/regnosys/rosetta/ide/server/RosettaRequestManager.java

```diff
@@ -35,7 +32,14 @@ public class RosettaRequestManager extends RequestManager {
 	                        .setDaemon(true)
 	                        .setNameFormat("rosetta-language-server-request-timeout-%d")
 	                        .build());
-		
+
+	/*
+	 * The code that uses this list fixes a memory leak in the RequestManager and should be contributed
+	 * back to the Xtext project then removed from here
+	 */
+	@VisibleForTesting
```

Comment:
The [docs for `VisibleForTesting`](https://guava.dev/releases/31.1-jre/api/docs/com/google/common/annotations/VisibleForTesting.html) say the following:

> *Do not use this interface* for public or protected declarations: it is a fig leaf for bad design, and it does not prevent anyone from using the declaration---and experience has shown that they will. If the method breaks the encapsulation of its class, then its internal representation will be hard to change. Instead, use [RestrictedApiChecker](http://errorprone.info/bugpattern/RestrictedApiChecker), which enforces fine-grained visibility policies.